### PR TITLE
How to use general message passing

### PR DESCRIPTION
**Pull request by krofax**
**Description**

This is a how-to guide with step by step instructions for how to use general message passing with interop(WIP)

**Tests**

**Additional context**

**Metadata**

Include a link to any github issues that this may close in the following form:
- Fixes - https://github.com/ethereum-optimism/devrel/issues/377

### CODE REVIEW

**Comment by krofax** [comment]
@blmalone @skeletor-spaceman This PR is ready for review.

**Review by brokewhale — COMMENTED** [review]
looks good

File: pages/stack/interop/message-passing.mdx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+3.  **Destination chain**:
+    *   Message receipt and validation
+    *   Execution of message content
+    *   Verification of successful processing
+
+## Implementation guide
+
+<Steps>
+  ### Prepare message sending
```
Comment (skeletor-spaceman) [inline pages/stack/interop/message-passing.mdx:78]: beware of bad formatting here

File: pages/stack/interop/message-passing.mdx [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+  The [CrossChainMessenger](https://github.com/ethereum-optimism/optimism/blob/92ed64e171c6eb9c6a080c626640e8836f0653cc/packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol) contract serves as the primary interface for cross-chain communication.
```
Comment (skeletor-spaceman) [inline pages/stack/interop/message-passing.mdx:80]:
```suggestion
  The [CrossDomainMessenger](https://github.com/ethereum-optimism/optimism/blob/92ed64e171c6eb9c6a080c626640e8836f0653cc/packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol) contract serves as the primary interface for cross-chain communication. 
```

File: pages/stack/interop/message-passing.mdx [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    *   Execution of message content
+    *   Verification of successful processing
+
+## Implementation guide
+
+<Steps>
+  ### Prepare message sending
+
+  The [CrossChainMessenger](https://github.com/ethereum-optimism/optimism/blob/92ed64e171c6eb9c6a080c626640e8836f0653cc/packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol) contract serves as the primary interface for cross-chain communication. 
+  It initializes the connection to the L2 messenger contract and provides the base functionality for sending messages across chains.
```
Comment (skeletor-spaceman) [inline pages/stack/interop/message-passing.mdx:81]: this is not correct, the CDM does not connect to any L2 messenger, it's a standalone sender that only emits an event.

File: pages/stack/interop/message-passing.mdx [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+          uint32 _gasLimit      // Gas limit for execution on destination
+      ) external {
+          IL2CrossDomainMessenger(L2_MESSENGER).sendMessage(
+              _target,
+              _message,
+              _gasLimit
+          );
+      }
+  }
+  ```
```
Comment (skeletor-spaceman) [inline pages/stack/interop/message-passing.mdx:104]: where was this taken out of? there is no need to set a `_minGasLimit`. I think a confusion was made with the L1<>L2 messengers.

File: pages/stack/interop/message-passing.mdx [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    const tx = await contract.sendMessage(
+      target,
+      message,
+      gasLimit,
+      { gasLimit: 200000 } // Transaction gas limit for sending
+    );
+    
+    return tx.hash;
+  }
+  ```
```
Comment (skeletor-spaceman) [inline pages/stack/interop/message-passing.mdx:132]: same as above, a confusion was made between L1<>L2 and the interop L2<>L2 messengers

**Review by skeletor-spaceman — CHANGES_REQUESTED** [review]

File: pages/stack/interop/message-passing.mdx (hunk 4 above)
Comment (krofax) [reply pages/stack/interop/message-passing.mdx:104]: @skeletor-spaceman What would be the correct code here, if we were to take out `_minGasLimit`?

File: pages/stack/interop/message-passing.mdx (hunk 5 above)
Comment (krofax) [reply pages/stack/interop/message-passing.mdx:132]: @skeletor-spaceman What will you advice here to improve the code?

File: pages/stack/interop/message-passing.mdx (hunk 4 above)
Comment (skeletor-spaceman) [reply pages/stack/interop/message-passing.mdx:104]: hey @krofax ! @zainbacchus will help on this :)

**Comment by krofax** [comment]
@zainbacchus Please can you review this when you have the time.

**Review by zainbacchus — APPROVED** [review]

File: pages/stack/interop/message-passing.mdx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    
+    Note over Source,Dest: Message Creation
+    Source->>Messenger: Emit message event
+    Note over Source,Dest: Message Serialization
+    Messenger-->>Messenger: Convert to standard format
+    Note over Source,Dest: Identifier Creation
+    Messenger-->>Messenger: Generate unique ID
+    Note over Source,Dest: Message Execution
+    Messenger->>Dest: Process message
+```
```
[the remark this answers is not in the document]
Comment (krofax) [reply pages/stack/interop/message-passing.mdx:39]: @coderabbitai generate as simplified sequence diagram to match actual implementation.

**Review by cpengilly — APPROVED** [review]

File: pages/stack/interop/_meta.json [hunk 7]
```diff
@@ -2,5 +2,6 @@
     "explainer": "Interop explainer",
     "cross-chain-message": "Anatomy of cross-chain message",
     "supersim": "Supersim Multichain Development Environment",
-    "superchain-erc20": "SuperchainERC20"
+    "superchain-erc20": "SuperchainERC20 token standard",
```
Comment (cpengilly) [inline pages/stack/interop/_meta.json:5]: just "SuperchainERC" (no token standard)

Comment (cpengilly) [inline pages/stack/interop/_meta.json:5]: just "SuperchainERC20" (no token standard)
```suggestion
    "superchain-erc20": "SuperchainERC20",
```

Comment (cpengilly) [reply pages/stack/interop/_meta.json:5]: we're not using 'token standard' per product team

**Comment by krofax** [comment]
@skeletor-spaceman Could you do a quick final review on your end? Zain has already approved.

**Review by skeletor-spaceman — APPROVED** [review]
matching zains approval

**Pull request merged**